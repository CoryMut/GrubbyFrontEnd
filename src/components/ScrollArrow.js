import React, { useState, useEffect, useCallback } from "react";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faArrowCircleUp } from "@fortawesome/free-solid-svg-icons";
import { Link } from "react-scroll";

import "./Scroll.css";

const ScrollArrow = () => {
    // Track whether the scroll arrow is needed.
    const [showScroll, setShowScroll] = useState(null);
    // Check the scroll state, re-memoize when scroll state changes.
    const checkScrollTop = useCallback(() => {
        const headerHeight = 400;

        if (!showScroll && window.pageYOffset > headerHeight) {
            setShowScroll(true);
        } else if (showScroll && window.pageYOffset <= headerHeight) {
            setShowScroll(false);
        }
    }, [showScroll]);
    // Add/remove the event listener when the component is unmounted or the scroll state has changed.
    useEffect(() => {
        window.addEventListener("scroll", checkScrollTop);
        return () => {
            window.removeEventListener("scroll", checkScrollTop);
        };
    }, [checkScrollTop]);

    // const scrollTop = () => {
    //     window.scrollTo({ top: 0, behavior: "smooth" });
    // };

    return (
        <div style={{ display: "flex", justifyContent: "center" }}>
            <div className="scrollTop">
                <Link activeClass="active" to="top" spy={true} smooth={true} offset={-70} duration={500}>
                    <FontAwesomeIcon
                        icon={faArrowCircleUp}
                        size="2x"
                        style={{ display: showScroll ? "flex" : "none", textAlign: "center" }}
                    />
                </Link>
            </div>
            {/* <div className="scrollTop">
                <FontAwesomeIcon
                    icon={faArrowCircleUp}
                    onClick={scrollTop}
                    size="2x"
                    style={{ display: showScroll ? "flex" : "none", textAlign: "center" }}
                />
            </div> */}
        </div>
    );
};

export default ScrollArrow;
