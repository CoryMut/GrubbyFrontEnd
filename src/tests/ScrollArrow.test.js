import React from "react";
import { shallow } from "enzyme";
import ScrollArrow from "../components/ScrollArrow";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router";

jest.mock("react-router-dom", () => ({
    useHistory: () => ({
        push: jest.fn(),
    }),
    useLocation: () => ({
        push: jest.fn(),
    }),
}));

test("should test NavBar component", () => {
    const wrapper = shallow(<ScrollArrow />);
    expect(wrapper).toMatchSnapshot();
});

test("it renders without crashing", () => {
    render(
        <MemoryRouter>
            <ScrollArrow />
        </MemoryRouter>
    );
});
