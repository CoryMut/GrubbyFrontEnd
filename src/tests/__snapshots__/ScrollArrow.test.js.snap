// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should test NavBar component 1`] = `
<div
  style={
    Object {
      "display": "flex",
      "justifyContent": "center",
    }
  }
>
  <div
    className="scrollTop"
  >
    <Link
      activeClass="active"
      duration={500}
      offset={-70}
      smooth={true}
      spy={true}
      to="top"
    >
      <FontAwesomeIcon
        border={false}
        className=""
        fixedWidth={false}
        flip={null}
        icon={
          Object {
            "icon": Array [
              512,
              512,
              Array [],
              "f0aa",
              "M8 256C8 119 119 8 256 8s248 111 248 248-111 248-248 248S8 393 8 256zm143.6 28.9l72.4-75.5V392c0 13.3 10.7 24 24 24h16c13.3 0 24-10.7 24-24V209.4l72.4 75.5c9.3 9.7 24.8 9.9 34.3.4l10.9-11c9.4-9.4 9.4-24.6 0-33.9L273 107.7c-9.4-9.4-24.6-9.4-33.9 0L106.3 240.4c-9.4 9.4-9.4 24.6 0 33.9l10.9 11c9.6 9.5 25.1 9.3 34.4-.4z",
            ],
            "iconName": "arrow-circle-up",
            "prefix": "fas",
          }
        }
        inverse={false}
        listItem={false}
        mask={null}
        pull={null}
        pulse={false}
        rotation={null}
        size="2x"
        spin={false}
        style={
          Object {
            "display": "none",
            "textAlign": "center",
          }
        }
        swapOpacity={false}
        symbol={false}
        title=""
        transform={null}
      />
    </Link>
  </div>
</div>
`;
